eduNEXT/frontend-lib-content-components
# feat: add grading strategy field in problem editor

## Description
This PR adds a new field in the Problem Editor for choosing a **Grading Strategy**. Currently, the only Grading Strategy is the Last Score. From now on, the new grading strategies are:

1. **Last Score _(Default)_**: The last score made is taken for grading.
2. **First Score**: The first score made is taken for grading.
3. **Highest Score**: The highest score made is taken for grading.
4. **Average Score**: The average of all scores made is taken for grading.

## Screenshots
![image](https://github.com/eduNEXT/frontend-lib-content-components/assets/64033729/c4c09b35-bd2d-40ec-a2fd-2fe9b3d9f29a)
![image](https://github.com/eduNEXT/frontend-lib-content-components/assets/64033729/5ffbaa67-552c-4644-bf8d-d0f83e18eabe)
![image](https://github.com/eduNEXT/frontend-lib-content-components/assets/64033729/9e6ba85d-e59f-4585-b99b-b50b5161f855)

## Dependencies
This PR needs the changes made from [this PR](https://github.com/eduNEXT/edx-platform/pull/299), which includes the new field from the Backend.

## Testing Instructions

Using `tutor` you can test these changes:

1. In your tutor environment clone the [frontend-app-course-authoring MFE](https://github.com/openedx/frontend-app-course-authoring/)
2. Create a mount with `tutor mounts add frontend-app-course-authoring`
3. Run `tutor config save` and `tutor dev start -d`
4. Inside the `frontend-app-course-authoring` folder clone this repository and checkout to this branch.
6. Inside the `frontend-app-course-authoring` folder create a new file called `module.config.js` with this content:

    ```javascript
    module.exports = {
      localModules: [
        {
          moduleName: "@edx/frontend-lib-content-components/dist",
          dir: "./frontend-lib-content-components",
          dist: "src",
        },
        {
          moduleName: "@edx/frontend-lib-content-components",
          dir: "./frontend-lib-content-components",
          dist: "src",
        },
      ],
    };
    ``` 
7. Then run `npm install`. First in `frontend-lib-content-components`, then in `frontend-app-course-authoring`.
8. Active in the Django admin (`admin/waffle/flag/`) this waffle flags: `new_core_editors.use_new_problem_editor` and `new_core_editors.use_new_text_editor` like this:

    ![image](https://github.com/eduNEXT/frontend-lib-content-components/assets/64033729/c0d986a1-5b95-46ac-936c-e7317688dbce)
9. Go to Studio and create a new component with a problem, for example: **Single select**
10. In the settings of the component you should see the new field.
